src/meta/processors/indexMan/CreateTagIndexProcessor.cpp

```diff
@@ -9,111 +9,172 @@
 namespace nebula {
 namespace meta {
 
-void CreateTagIndexProcessor::process(const cpp2::CreateTagIndexReq& req) {
-    auto space = req.get_space_id();
-    CHECK_SPACE_ID_AND_RETURN(space);
-    const auto &indexName = req.get_index_name();
-    auto &tagName = req.get_tag_name();
-    auto &fieldNames = req.get_fields();
-    if (fieldNames.empty()) {
-        LOG(ERROR) << "The index field of an tag should not be empty.";
-        handleErrorCode(cpp2::ErrorCode::E_INVALID_PARM);
-        onFinished();
-        return;
-    }
-    std::set<std::string> columnSet(fieldNames.begin(), fieldNames.end());
-    if (fieldNames.size() != columnSet.size()) {
+cpp2::ErrorCode CreateTagIndexProcessor::checkFields() {
+    std::unordered_set<std::string> columnSet(fieldNames_.begin(), fieldNames_.end());
+    if (fieldNames_.size() != columnSet.size()) {
         LOG(ERROR) << "Conflict field in the tag index.";
-        handleErrorCode(cpp2::ErrorCode::E_CONFLICT);
-        onFinished();
-        return;
+        return cpp2::ErrorCode::E_CONFLICT;
     }
 
-    // A maximum of 16 columns are allowed in the index.
-    if (columnSet.size() > 16) {
-        LOG(ERROR) << "The number of index columns exceeds maximum limit 16";
-        handleErrorCode(cpp2::ErrorCode::E_CONFLICT);
-        onFinished();
-        return;
-    }
-
-    folly::SharedMutex::WriteHolder wHolder(LockUtils::tagIndexLock());
-    auto ret = getIndexID(space, indexName);
-    if (ret.ok()) {
-        LOG(ERROR) << "Create Tag Index Failed: " << indexName << " have existed";
-        if (req.get_if_not_exists()) {
-            handleErrorCode(cpp2::ErrorCode::SUCCEEDED);
-        } else {
-            handleErrorCode(cpp2::ErrorCode::E_EXISTED);
+    switch (indexType_) {
+        case cpp2::IndexType::NORMAL: {
+            if (fieldNames_.empty()) {
+                LOG(ERROR) << "The normal index field of a tag should not be empty.";
+                return cpp2::ErrorCode::E_INVALID_PARM;
+            }
+            // A maximum of 16 columns are allowed in the normal index.
+            if (fieldNames_.size() > 16) {
+                LOG(ERROR) << "The number of normal index columns exceeds maximum limit 16";
+                return cpp2::ErrorCode::E_CONFLICT;
+            }
+            break;
         }
-        onFinished();
-        return;
+        case cpp2::IndexType::VERTEX: {
+            // No columns are allowed.
+            if (fieldNames_.size() > 0) {
+                LOG(ERROR) << "No columns are allowed in the vertex index";
```

Comment:
Can't quite follow the logic here, why when then filed name size is greater than 0, no columns are allowed?